src/main/java/com/redhat/rhjmc/containerjfr/net/SslConfiguration.java

```diff
@@ -0,0 +1,207 @@
+package com.redhat.rhjmc.containerjfr.net;
+
+import com.redhat.rhjmc.containerjfr.core.log.Logger;
+import com.redhat.rhjmc.containerjfr.core.sys.Environment;
+import com.redhat.rhjmc.containerjfr.core.sys.FileSystem;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PfxOptions;
+import org.openjdk.jmc.common.util.Pair;
+
+import java.nio.file.Path;
+
+class SslConfiguration {
+    private final Environment env;
+    private final FileSystem fs;
+    private final Logger logger;
+
+    private final Path keystorePath;
+    private final String keystorePass;
+
+    private final Path keyPath;
+    private final Path certPath;
+
+    private static final String KEYSTORE_PATH_ENV = "KEYSTORE_PATH";
+    private static final String KEYSTORE_PASS_ENV = "KEYSTORE_PASS";
+    private static final String KEY_PATH_ENV = "KEY_PATH";
+    private static final String CERT_PATH_ENV = "CERT_PATH";
+
+    SslConfiguration(Environment env, FileSystem fs, Logger logger) throws IllegalArgumentException {
+        this.env = env;
+        this.fs = fs;
+        this.logger = logger;
+
+        {
+            Path path = obtainKeyStorePathIfSpecified();
+            if (path != null) {
+                keystorePath = path;
+                keystorePass = env.getEnv(KEYSTORE_PASS_ENV, "");
+                keyPath = null;
+                certPath = null;
+                return;
+            }    
+        }
+
+        {
+            Pair<Path, Path> pair = obtainKeyCertPathPairIfSpecified();
+            if (pair != null) {
+                keystorePath = null;
+                keystorePass = null;
+                keyPath = pair.left;
+                certPath = pair.right;
+                return;
+            }
+        }
+
+        {
+            Path path = discoverKeyStorePathInDefaultLocations();
+            if (path != null) {
+                keystorePath = path;
+                keystorePass = env.getEnv(KEYSTORE_PASS_ENV, "");
+                keyPath = null;
+                certPath = null;
+                return;
+            }    
+        }
+
+        {
+            Pair<Path, Path> pair = discoverKeyCertPathPairInDefaultLocations();
+            if (pair != null) {
+                keystorePath = null;
+                keystorePass = null;
+                keyPath = pair.left;
+                certPath = pair.right;
+                return;
+            }
+        }
+
+        keystorePath = null;
+        keystorePass = null;
+        keyPath = null;
+        certPath = null;
+        logger.warn("No available SSL certificates. Fallback to plain HTTP.");
+    }
+
+    Path obtainKeyStorePathIfSpecified() throws IllegalArgumentException {
+        if (!env.hasEnv(KEYSTORE_PATH_ENV)) {
+            return null;
+        }
+
+        Path path = Path.of(env.getEnv(KEYSTORE_PATH_ENV)).normalize();
+        if (!fs.exists(keystorePath)) {
```

Comment:
I think we're missing a corresponding PR to -core to add this `exists` method.

Reply:
Thanks for the reminder. I have a PR ready and will submit once ready.